azure-pipelines.yml

```diff
@@ -137,28 +136,51 @@ steps:
   - task: Docker@2
     displayName: Login to temporary pipeline ACR
     inputs:
-      command: login
-      containerRegistry: $(PIPELINE_CONTAINER_REGISTRY)
+      containerRegistry: 'ASO Pipeline ACR'
+      command: 'login'
 
   - task: Docker@2
     displayName: Build and Push Docker Image to temporary ACR for validation
     inputs:
-      command: buildAndPush
-      containerRegistry: $(PIPELINE_CONTAINER_REGISTRY)
-      repository: $(IMAGE_NAME)
-      Dockerfile: "$(System.DefaultWorkingDirectory)/Dockerfile"    
-      buildContext: $(System.DefaultWorkingDirectory)
-      tags: |
-        $(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)
+      containerRegistry: 'ASO Pipeline ACR'
+      repository: '$(IMAGE_NAME)'
+      command: 'buildAndPush'
+      Dockerfile: '$(System.DefaultWorkingDirectory)/Dockerfile'
+      buildContext: '$(System.DefaultWorkingDirectory)'
+      tags: '$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)'
+
+#  - task: Kubernetes@1
+#    displayName: Deploy to AKS - Fetch kubeconfig
+#    inputs:
+#      connectionType: 'Kubernetes Service Connection'
+#      kubernetesServiceEndpoint: 'ASO-AKS'
+#      namespace: $(OPERATOR_NAMESPACE)
+#      command: 'login'
+#      checkLatest: true
 
-  - task: Kubernetes@1
-    displayName: Deploy to AKS - Fetch kubeconfig
+  
+  - task: AzureCLI@2
     inputs:
-      connectionType: 'Kubernetes Service Connection'
-      kubernetesServiceEndpoint: 'ASO-AKS'
-      namespace: $(OPERATOR_NAMESPACE)
-      command: 'login'
-      checkLatest: true
+      azureSubscription: 'ASO Subscription'
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: |
+        az resource list --tag freeforpipeline=true --query '[].{Name:name}' -o table > clusters.txt
+        clustername=$(tail -n 1 clusters.txt)
+        if [ "$clustername" == "" ]
+        then
+        echo AKS clusters unavailable
+        exit 1  
+        fi
+        echo $clustername
+        echo "##vso[task.setvariable variable=chosenclustername]$clustername"
+        echo 'az aks get-credentials -g $(AKS_CLUSTER_RG) -n $clustername'
+        az aks get-credentials -g $(AKS_CLUSTER_RG) -n $clustername
+        # Set tags to not available for the selected cluster so it doesn't get used
+        az resource tag --tags 'freeforpipeline=false' -g $(AKS_CLUSTER_RG) -n $clustername --resource-type Microsoft.ContainerService/managedClusters
+      workingDirectory: '$(System.DefaultWorkingDirectory)'
```

Comment:
You don't really need to set this because the working directory is set to this for each task by default.

Reply:
Ok, I left it as is to make sure there isnt any new issues